app/src/main/java/com/example/tasky/common/domain/TextValidatorExtension.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.tasky.common.domain
+
+import android.util.Patterns
+
+
+fun String.isValidPassword(): Boolean {
+    val passwordPattern = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).{9,}$"
+    return this.matches(passwordPattern.toRegex())
+}
```

Comment:
For simple patterns like a valid password, I'm always a friend of doing this manually since it lets you quickly know what makes a valid password. Something like this: 
```
val hasLowerCase = password.any { it.isLowerCase() }
val hasUpperCase = password.any { it.isUpperCase() }
val hasDigit = password.any { it.hasDigit() }
val hasValidLength = password.length in min..max

return hasLowerCase && hasUpperCase && hasDigit && hasValidLength
```
That way, probably even my grandma could tell you what the requirements for a valid password are :D 
Regex makes sense if you can't easily do this manually, but has the cost of making code very unreadable.